docs/aws/audit/lambdamonitoring/rules/has_tags_remediation.mdx

```diff
@@ -1,6 +1,293 @@
+               else:
+                   print(f"All required tags are present for {function_name}")
+
+       except ClientError as e:
+           print(f"Error enforcing tags on existing Lambda functions: {e}")
+
+   # Enforce tags on existing functions
+   enforce_tags_on_existing_functions()
+   ```
+
+4. **Automate Tagging Enforcement:**
+   Set up a scheduled AWS Lambda function or a CloudWatch rule to periodically run the script to ensure compliance.
+
+   ```python
+   import boto3
+
+   def lambda_handler(event, context):
+       enforce_tags_on_existing_functions()
+
+   # Deploy this function and set up a CloudWatch rule to trigger it periodically
+   ```
+
+By following these steps, you can ensure that all your AWS Lambda functions are created with the required tags and that existing functions are checked and updated accordingly.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Lambda service by typing 'Lambda' into the search bar and selecting 'Lambda' from the dropdown menu.
+3. Once you are in the AWS Lambda console, you will see a list of all your Lambda functions. Click on the name of the function you want to check for tags.
+4. In the function's detail page, scroll down to the 'Tags' section. If there are no tags listed, then the Lambda function does not have any tags.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all your Lambda functions. Open your terminal and run the following AWS CLI command:
+
+```bash
+aws lambda list-functions --region your-region
+```
+Replace "your-region" with your actual AWS region. This command will return a list of all your Lambda functions in the specified region.
+
+2. The output will be in JSON format. Each function will have an "FunctionName" field. You can use this field to get more information about each function. 
+
+3. To check if a function has tags, use the following command:
+
+```bash
+aws lambda list-tags --resource your-function-arn
+```
+Replace "your-function-arn" with the actual ARN of your Lambda function. This command will return a list of all tags associated with the specified function.
+
+4. If the output is empty, it means the function doesn't have any tags. If the output contains key-value pairs, these are the tags of your function. Repeat steps 3 and 4 for each function in your list.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. This will allow you to interact with AWS services including Lambda. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>